implementation/src/main/java/io/smallrye/jwt/auth/AbstractDpopTokenValidator.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.smallrye.jwt.auth;
```

Comment:
Would it make sense to keep DPoP in `auth.dpop` ? If we will get more than one class then it might work well longer term, not  a big deal, just a question/suggestion

Reply:
Using a separate package for this is a good idea. At the very least, there will be the Logging/Message interfaces. I will rework this into its own `auth.dpop` package.